torchgeo/datasets/deepglobelandcover.py

```diff
@@ -0,0 +1,267 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""DeepGlobe Land Cover Classification Challenge dataset."""
+
+import os
+from typing import Callable, Dict, Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from matplotlib.figure import Figure
+from PIL import Image
+from torch import Tensor
+
+from .geo import VisionDataset
+from .utils import (
+    check_integrity,
+    draw_semantic_segmentation_masks,
+    extract_archive,
+    rgb_to_mask,
+)
+
+
+class DeepGlobeLandCover(VisionDataset):
+    """DeepGlobe Land Cover Classification Challenge dataset.
+
+    The `DeepGlobe Land Cover Classification Challenge
+    <https://competitions.codalab.org/competitions/18468/>`__ dataset
+    offers high-resolution sub-meter satellite imagery focusing for the task of
+    semantic segmentation to detect areas of urban, agriculture, rangeland, forest,
+    water, barren, and unknown. It contains 1,146 satellite images of size
+    2448 x 2448 pixels in total, split into training/validation/test sets, the original
+    dataset can be downloaded from `Kaggle <https://www.kaggle.com/datasets/balraj98/
+    deepglobe-land-cover-classification-dataset>`__.
+    However, we only use the training dataset with 803 images since the original test
+    and valid dataset are not accompanied by labels. The dataset that we use with a
+    custom train/test split can be downloaded from `Kaggle <https://www.kaggle.com/
+    datasets/geoap96/deepglobe2018-landcover-segmentation-traindataset>`__ (created as a
```

Comment:
Do URLs work when split onto multiple lines like this? Need to check the generated docs...

Reply:
Looks like they do, cool.